src/ir_Trotec.cpp

```diff
@@ -0,0 +1,170 @@
+#include "ir_Trotec.h"
+#include <algorithm>
+#include "IRremoteESP8266.h"
+#include "IRutils.h"
+
+// Constants
+#define TROTEC_HDR_MARK   5952U
+#define TROTEC_HDR_SPACE  7364U
+#define TROTEC_ONE_MARK    592U
+#define TROTEC_ONE_SPACE  1560U
+#define TROTEC_ZERO_MARK   592U
+#define TROTEC_ZERO_SPACE  592U
+#define TROTEC_GAP        6184U
+
+
+#if SEND_TROTEC
+
+void IRsend::sendTrotec(unsigned char data[], uint16_t nbytes,
+                        uint16_t repeat) {
+  if (nbytes < TROTEC_COMMAND_LENGTH)
+    return;
+
+  enableIROut(36);
+
+  for (uint16_t r = 0; r <= repeat; r++) {
+    // Header
+    mark(TROTEC_HDR_MARK);
+    space(TROTEC_HDR_SPACE);
+
+    // Data
+    for (uint16_t i = 0; i < nbytes; i++)
+      sendData(TROTEC_ONE_MARK, TROTEC_ONE_SPACE, TROTEC_ZERO_MARK,
+               TROTEC_ZERO_SPACE, data[i], 8, false);
+
+    // Footer
+    mark(TROTEC_ONE_MARK);
+    space(TROTEC_GAP);
+    mark(TROTEC_ONE_MARK);
+    ledOff();
+  }
+}
+
+IRTrotecESP::IRTrotecESP(uint16_t pin) : _irsend(pin) {
+  stateReset();
+}
+
+void IRTrotecESP::begin() {
+  _irsend.begin();
+}
+
+void IRTrotecESP::send() {
+  _irsend.sendTrotec(trotec);
+}
+
+void IRTrotecESP::checksum() {
+  uint8_t sum = 0;
+  uint8_t i;
+
+  for (i = 2; i < 8; i++) sum += trotec[i];
+
+  trotec[8] = sum & 0xFF;
+}
+
+
+void IRTrotecESP::stateReset() {
+  for (uint8_t i = 2; i < TROTEC_COMMAND_LENGTH; i++)
+    trotec[i] = 0x0;
+
+  trotec[0] = TROTEC_INTRO1;
+  trotec[1] = TROTEC_INTRO2;
+
+  setPower(0);
```

Comment:
Nitpick:
It would be great if you used constants for these values below.
e.g.
What is _0_ in this context? I think you should be using `true` and `false` given it's a Boolean.
What is speed _2_ etc?

Maybe something like this would make things clearer:
```
#define TROTEC_DEF_TEMP  25
...
setTemp(TROTEC_DEF_TEMP);
```